Userland/Libraries/LibJS/Bytecode/Generator.cpp

```diff
@@ -262,32 +262,73 @@ CodeGenerationErrorOr<void> Generator::emit_delete_reference(JS::ASTNode const&
     return {};
 }
 
-Label Generator::perform_needed_unwinds_for_labelled_break_and_return_target_block(DeprecatedFlyString const& break_label)
+void Generator::generate_break()
+{
+    bool last_was_finally = false;
+    // FIXME: Reduce code duplication
+    for (size_t i = m_boundaries.size(); i > 0; --i) {
+        auto boundary = m_boundaries[i - 1];
+        using enum BlockBoundaryType;
+        switch (boundary) {
+        case Break:
+            emit<Op::Jump>().set_targets(nearest_breakable_scope(), {});
+            return;
+        case Unwind:
+            if (!last_was_finally)
+                emit<Bytecode::Op::LeaveUnwindContext>();
+            last_was_finally = false;
+            break;
+        case LeaveLexicalEnvironment:
+            emit<Bytecode::Op::LeaveEnvironment>(Bytecode::Op::EnvironmentMode::Lexical);
+            break;
+        case LeaveVariableEnvironment:
+            emit<Bytecode::Op::LeaveEnvironment>(Bytecode::Op::EnvironmentMode::Var);
+            break;
+        case Continue:
+            break;
+        case ReturnToFinally: {
+            auto& block = make_block(DeprecatedString::formatted("{}.break", current_block().name()));
+            emit<Op::ScheduleJump>(Label { block });
```

Comment:
Tangentially related sidenote: .NET on x86_64 generates a _call_ for finally, that's kinda smort - maybe we could add a `CallBasicBlock` insn to do the same, it's a nicer mechanism imo.

Reply:
I added this and some other notes to the `ScheduleJump` instruction for now